src/tqec/compile/tree/annotators/detectors.py

```diff
@@ -198,7 +194,7 @@ def __init__(
             )
         self._k = k
         self._manhattan_radius = manhattan_radius
-        self._database = detector_database
+        self._database = detector_database if detector_database is not None else DetectorDatabase()
```

Comment:
When reviewing, could you double-check that change? I cannot remember why such a change was made, it contradicts the docstring, and so I would lean on the side of undoing it, but want a double-check before doing so.

Reply:
Let's undo it then as a value of `None` carries a special meaning that we might need in the called functions.